perfkitbenchmarker/benchmarks/hbase_ycsb_benchmark.py

```diff
@@ -0,0 +1,216 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Runs YCSB against HBase.
+
+
+HBase is a scalable NoSQL database built on Hadoop.
+https://hbase.apache.org/
+
+A running installation consists of:
+  * An HDFS NameNode.
+  * HDFS DataNodes.
+  * An HBase master node.
+  * HBase regionservers.
+  * A zookeeper cluster (https://zookeeper.apache.org/).
+
+See: http://hbase.apache.org/book.html#_distributed.
+
+This benchmark provisions:
+  * A single node functioning as HDFS NameNode, HBase master, and zookeeper
+    quorum member.
+  * '--num_vms - 1' nodes serving as both HDFS DataNodes and HBase region
+    servers (so region servers and data are co-located).
+By default only the master node runs Zookeeper. Some regionservers may be added
+to the zookeeper quorum with the --hbase_zookeeper_nodes flag.
+
+
+HBase web UI on 15030.
+HDFS web UI on  50070.
+"""
+
+import logging
+import os
+
+from perfkitbenchmarker import data
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.packages import hadoop
+from perfkitbenchmarker.packages import hbase
+from perfkitbenchmarker.packages import ycsb
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_integer('hbase_zookeeper_nodes', 1, 'Number of Zookeeper nodes.')
+flags.DEFINE_boolean('hbase_use_snappy', True,
+                     'Whether to use snappy compression.')
+
+BENCHMARK_INFO = {'name': 'hbase_ycsb',
+                  'description': 'Run YCSB against HBase. Specify the HBase '
+                  'cluster size with --num_vms. Specify the number of YCSB VMs '
+                  'with --ycsb_client_vms.',
+                  'scratch_disk': True,
+                  'num_machines': None}
+
+HBASE_SITE = 'hbase-site.xml'
+
+CREATE_TABLE_SCRIPT = 'hbase/create-ycsb-table.hbaseshell.j2'
+TABLE_NAME = 'usertable'
+COLUMN_FAMILY = 'cf'
+TABLE_SPLIT_COUNT = 200
+
+
+def GetInfo():
+  info = BENCHMARK_INFO.copy()
+  num_vms = max(FLAGS.num_vms, 2)
+  if FLAGS['num_vms'].present and FLAGS.num_vms < 2:
+    raise ValueError('hbase_ycsb requires at least 2 HBase VMs.')
+  info['num_machines'] = num_vms + FLAGS.ycsb_client_vms
+  return info
+
+
+def CheckPrerequisites():
+  """Verifies that the required resources are present.
+
+  Raises:
+    perfkitbenchmarker.data.ResourceNotFound: On missing resource.
+  """
+  hbase.CheckPrerequisites()
+  hadoop.CheckPrerequisites()
+  ycsb.CheckPrerequisites()
+
+
+def _InstallYCSB(vm):
```

Comment:
any reason why InstallHbase() is nested next to runthreaded and this is defined at the top level?